MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -1150,18 +1151,50 @@ public Unit addUnit(Entity en, boolean allowNewPilots, int days) {
         return units.values();
     }
 
-    public ArrayList<Unit> getUnits(boolean weightSorted, boolean alphaSorted) {
-        ArrayList<Unit> sortedUnits = getCopyOfUnits();
-        if (alphaSorted || weightSorted) {
-            if (alphaSorted) {
-                sortedUnits.sort(Comparator.comparing(Unit::getName));
-            }
-            if (weightSorted) {
+    public List<Unit> getUnits(boolean weightSorted) {
+        return getUnits(false, weightSorted, false);
+    }
+
+    /**
+     * This returns a list of the current units, sorted alphabetically and potentially by other methods
+     * @param weightSorted      True if the unit list is sorted by weight descending, otherwise false
+     * @param weightClassSorted True if the unit list is sorted by weight class in format heaviest to lightest, otherwise false
+     * @param unitTypeSorted    True if the unit list is sorted by the unit type
+     * @return a copy of the units in the campaign with the applicable sort format
+     */
+    public List<Unit> getUnits(boolean weightClassSorted, boolean weightSorted,
```

Comment:
Are these mutually exclusive or "then by"?

Reply:
Currently, they are applied as a then by, although you can do the first or the second alone. I'm going to replace it with an enum-based sorter method to handle all the possible use cases. I'm 99% code done for that, it just needs to be updated with a few of the comments here.

Reply:
If that's the case, then this would be easier to understand if you chained the comparators:

```java
Comparator<Unit> comparator = Comparator.comparing(u -> u.getName(), new NaturalOrderComparator());
...
if (expr) {
    comparator = comparator.thenComparing(...);
}

if (otherExpr) {
    comparator = comparator.thenComparing(...);
}

units.sort(comparator);
```

Reply:
I'm opening a PR to replace this either later today or tomorrow with the enum (didn't want to distract from the main portion of this. Is it okay if I just leave it like this, and implement it in a final case then?